### PR TITLE
Add early stopping on test set for LearnerSurvXgboost

### PR DESCRIPTION
**Pull request by bblodfon**
# Updated Learner

This pull request updates LearnerSurvXgboost.

### Reason for update

See issue https://github.com/mlr-org/mlr3extralearners/issues/250

## Dependencies

I have...

* [ ] Added more than one dependency to suggests - If ticked, explain why below.
* [x] Added no dependencies to imports.
* [ ] Added dependencies to remotes - If ticked, explain why below.

## Tests

* [x] Learner autotest is passing locally.
* [x] Paramtest is passing locally.
* [ ] I have run `rcmdcheck::rcmdcheck` and there are no new NOTEs, WARNINGs, or ERRORs.

## Cleaning

I have run...

* [x] `devtools::document(roclets = c('rd', 'collate', 'namespace'))`

If you've also made changes other than a small patch (i.e. changes that could affect code style) then please also confirm:

I have run...

* [x] `styler.mlr::style_pkg()` - If needed install with `remotes::install_github('mlr-org/styler.mlr')`
* [ ] `lintr::lint_package()`

And

* [ ] Updated NEWS.md
* [ ] Updated the package version (For patches (bug fixing, small added feature): add 1 to the 3rd number. For minor updates (new learner, new features): add 1 to the 2nd number).

### CODE REVIEW

**Comment by sebffischer** [comment]
sorry I am on vacation right now but will get to it soon I hope :)

**Comment by sebffischer** [comment]
thank you already!

**Comment by bblodfon** [comment]
Hi @sebffischer! Just a kindly remainder to check this pull request, I would really like to get this feature properly implemented and used in my benchmarks soon :)

**Comment by sebffischer** [comment]
Sorry for the delay and thanks for the reminder, I just needed some time off during my vacation!

**Comment by sebffischer** [comment]
Note that I opened an issue here https://github.com/mlr-org/mlr3learners/issues/257 to make it consistent with the other xgboost implementations.

**Comment by sebffischer** [comment]
Please include `mlr3pipelines` in Suggests in the DESCRIPTION

**Comment by bblodfon** [comment]
Ok, added the commits! Btw, I agree that it has become a little complicated in the code right now (order of elements and which are included in the end in the `watchlist`). Removing the early stopping on the train data itself will surely make it more readable.

**Review by sebffischer — COMMENTED** [review]
I have only one minor refactoring request.

**Comment by sebffischer** [comment]
I will merge and do the refactor myself to not let you wait any longer!
Thanks again for the contribution.

**Comment by bblodfon** [comment]
wait, almost had it!

**Comment by bblodfon** [comment]
I made a function called `.get_data` inside the `private` list, is that the correct way to do this? I getting an error like `could not find function ".get_data"`

**Comment by bblodfon** [comment]
okay found it! Let me know of stylistic/names changes that might be required

File: R/learner_xgboost_surv_xgboost.R [hunk 1]
```diff
@@ -151,43 +161,65 @@ delayedAssign(
...
-        if (is.null(pv$objective)) {
-          pv$objective = "survival:cox"
-        }
-
-        data = task$data(cols = task$feature_names)
-        target = task$data(cols = task$target_names)
+      # helper function to construct an `xgb.DMatrix` object
+      .get_data = function(task, pv, row_ids) {
+        # use all task rows if `rows_ids` is not specified
+        if (missing(row_ids))
```
Comment (sebffischer) [inline R/learner_xgboost_surv_xgboost.R:167]: don't use `missing()`, set default to `NULL` and use `if (is.null(row_ids))`

Comment (sebffischer) [reply R/learner_xgboost_surv_xgboost.R:167]: Although it is not so important here, this is generally a good practice. Otherwise it is not understandable from the function signature that the function has a default value.

**Review by sebffischer — COMMENTED** [review]
one more nitpick :)

File: R/learner_xgboost_surv_xgboost.R (hunk 1 above)
Comment (bblodfon) [reply R/learner_xgboost_surv_xgboost.R:167]: Thanks for the tip!

**Review by sebffischer — COMMENTED** [review]
lgtm!
